src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -14,211 +24,3020 @@ namespace Microsoft.CodeAnalysis.CSharp
+        /// than one location in the method being analyzed, then the method is partial and we prefer
+        /// to report an out parameter in partial method error.
+        /// </summary>
+        /// <param name="location">location to be used</param>
+        /// <returns>true if the out parameters of the method should be analyzed</returns>
+        protected bool ShouldAnalyzeOutParameters(out Location location)
+        {
+            var method = _member as MethodSymbol;
+            if ((object)method == null || method.Locations.Length != 1)
+            {
+                location = null;
+                return false;
             }
             else
             {
-                VisitCatchBlock(catchBlock, ref finallyState);
+                location = method.Locations[0];
+                return true;
             }
         }
 
-        private void VisitFinallyBlockWithUnassignments(BoundStatement finallyBlock, ref TLocalState unsetInFinally)
+        /// <summary>
+        /// Return the flow analysis state associated with a label.
+        /// </summary>
+        /// <param name="label"></param>
+        /// <returns></returns>
+        protected virtual TLocalState LabelState(LabelSymbol label)
         {
-            if (_trackUnassignments)
+            TLocalState result;
+            if (_labels.TryGetValue(label, out result))
             {
-                Optional<TLocalState> oldTryState = _tryState;
-                _tryState = AllBitsSet();
-                VisitFinallyBlock(finallyBlock, ref unsetInFinally);
-                var tempTryStateValue = _tryState.Value;
-                IntersectWith(ref unsetInFinally, ref tempTryStateValue);
-                if (oldTryState.HasValue)
-                {
-                    var oldTryStateValue = oldTryState.Value;
-                    IntersectWith(ref oldTryStateValue, ref tempTryStateValue);
-                    oldTryState = oldTryStateValue;
-                }
+                return result;
+            }
 
-                _tryState = oldTryState;
+            result = UnreachableState();
+            _labels.Add(label, result);
+            return result;
+        }
+
+        /// <summary>
+        /// Return to the caller the set of pending return statements.
+        /// </summary>
+        /// <returns></returns>
+        protected virtual ImmutableArray<PendingBranch> RemoveReturns()
+        {
+            ImmutableArray<PendingBranch> result;
+            if (_trackExceptions)
+            {
+                // when we are tracking exceptions, we use pendingBranches[0] to
+                // track exception states.
+                result = PendingBranches
+                        .Where(b => b.Branch != null)
+                        .AsImmutableOrNull();
+
+                var oldExceptions = PendingBranches[0];
+                Debug.Assert(oldExceptions.Branch == null);
```

Comment:
could use a comment.

Reply:
All the code under `if (_trackExceptions)` should be deleted.

---
In reply to: [236569305](https://github.com/dotnet/roslyn/pull/31389#discussion_r236569305) [](ancestors = 236569305)